development/ts-migration-dashboard/app/components/Connections.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import type { BoxModel } from './types';
+
+function buildShapePoints(coordinates: [number, number][]): string {
+  return coordinates.map(([x, y]) => `${x},${y}`).join(' ');
+}
+
+function buildPathD(coordinates: [number, number][]): string {
+  return coordinates
+    .map(([x, y], index) => {
+      if (index === 0) {
+        return `M ${x},${y}`;
+      }
+      return `L ${x},${y}`;
+    })
+    .join(' ');
+}
+
+function Arrowhead({
+  type,
+  x,
+  y,
+}: {
+  type: 'dependency' | 'dependent';
+  x: number;
+  y: number;
+}) {
+  return (
+    <polygon
+      className={`module-connection__${type}-arrowhead`}
+      points={buildShapePoints([
+        [x - 6, y - 6],
+        [x + 6, y - 6],
+        [x, y],
+      ])}
+    />
+  );
+}
+
+function Line({
+  type,
+  originX,
+  originY,
+  originYOffset = 0,
+  destinationX,
+  destinationY,
+  destinationYOffset = 0,
+}: {
+  type: 'dependency' | 'dependent';
+  originX: number;
+  originY: number;
+  originYOffset?: number;
+  destinationX: number;
+  destinationY: number;
+  destinationYOffset?: number;
+}) {
+  const coordinates: [number, number][] =
+    type === 'dependency'
+      ? [
+          [originX, originY],
+          [originX, originY + originYOffset],
+          [destinationX, originY + originYOffset],
+          [destinationX, destinationY],
+        ]
+      : [
+          [originX, originY],
+          [originX, destinationY - destinationYOffset],
+          [destinationX, destinationY - destinationYOffset],
+          [destinationX, destinationY],
+        ];
+  return (
+    <path
+      className={`module-connection__${type}`}
+      d={buildPathD(coordinates)}
+    />
+  );
+}
+
+function LineStart({
+  type,
+  x,
+  y,
+}: {
+  type: 'dependency' | 'dependent';
+  x: number;
+  y: number;
+}) {
+  return (
+    <circle className={`module-connection__${type}-point`} cx={x} cy={y} />
+  );
+}
+
+export default function Connections({ activeBox }: { activeBox: BoxModel }) {
```

Comment:
While the boxes are rendered on screen using one `div` per box, the connection lines for a particular box are drawn all in one element using SVG.